src/interp/interp.cc

```diff
@@ -1174,28 +1158,28 @@ RunResult Thread::StepInternal(Trap::Ptr* out_trap) {
       break;
     }
 
-    case O::I32Load:    return DoLoad<u32>(instr, out_trap);
```

Comment:
Maybe certain case statements like this one that use vertical alignment can be wrapped in "off" and we can leave the default to be multi-line.   Its kind of shame we can't configure it to use leave swtich/cases as single-line if and only if they are already, but I guess that isn't an option?

Reply:
I don't think we can tell clang-tidy to keep some part of the code if certain conditions are met.. 

Wrapping this part is not a problem, but this file has several similar parts:
https://github.com/WebAssembly/wabt/blob/b97dd591fad3e4b7d382d409247007cf65fc2baf/src/interp/interp.cc#L605-L610
https://github.com/WebAssembly/wabt/blob/b97dd591fad3e4b7d382d409247007cf65fc2baf/src/interp/interp.cc#L756-L762
https://github.com/WebAssembly/wabt/blob/b97dd591fad3e4b7d382d409247007cf65fc2baf/src/interp/interp.cc#L795-L801
https://github.com/WebAssembly/wabt/blob/b97dd591fad3e4b7d382d409247007cf65fc2baf/src/interp/interp.cc#L2416-L2424
https://github.com/WebAssembly/wabt/blob/b97dd591fad3e4b7d382d409247007cf65fc2baf/src/interp/interp.cc#L2427-L2439
https://github.com/WebAssembly/wabt/blob/b97dd591fad3e4b7d382d409247007cf65fc2baf/src/interp/interp.cc#L1177-L1200
...

Excluding every single one of these will not be very clean either. The switch-case you pointed is very big, so do you think wrapping only that one sounds OK?

I personally prefer fewer exceptions, especially for normal code. What I think exceptions are most warranted for is comments that don't make sense with 80-col limits, such as this one: https://github.com/WebAssembly/binaryen/blob/6a2d7f989065820476268a2382db2eccf72aadd7/src/support/safe_integer.cpp#L121-L145

Reply:
I think that sounds reasonable.

Reply:
Added `// clang-format off` and `on` to wrap this big switch-case. Left others as they are.